packages/network-controller/tests/NetworkController.test.ts

```diff
@@ -3496,6 +3496,447 @@ describe('NetworkController', () => {
     });
   });
 
+  describe('resetConnection', () => {
+    [NetworkType.mainnet, NetworkType.goerli, NetworkType.sepolia].forEach(
+      (networkType) => {
+        describe(`when the type in the provider configuration is "${networkType}"`, () => {
+          it('sets isCustomNetwork in state to false', async () => {
+            const messenger = buildMessenger();
+            await withController(
+              {
+                messenger,
+                state: {
+                  isCustomNetwork: true,
+                },
+                infuraProjectId: 'infura-project-id',
+              },
+              async ({ controller }) => {
+                const fakeInfuraProvider = buildFakeInfuraProvider();
+                createInfuraProviderMock.mockReturnValue(fakeInfuraProvider);
+                const fakeInfuraSubprovider = buildFakeInfuraSubprovider();
+                SubproviderMock.mockReturnValue(fakeInfuraSubprovider);
+                const fakeMetamaskProvider = buildFakeMetamaskProvider();
+                createMetamaskProviderMock.mockReturnValue(
+                  fakeMetamaskProvider,
+                );
+
+                await waitForStateChanges(messenger, {
+                  propertyPath: ['isCustomNetwork'],
+                  produceStateChanges: () => {
+                    controller.resetConnection();
+                  },
+                });
+
+                expect(controller.state.isCustomNetwork).toBe(false);
+              },
+            );
+          });
+
+          it('updates networkDetails.isEIP1559Compatible in state based on the latest block (assuming that the request for eth_getBlockByNumber is made successfully)', async () => {
```

Comment:
Nit: It might be good to put this test below "initializes a new provider object pointed to the current Infura network" so it's easier to compare with the tests for other methods.